logsink.c

```diff
@@ -0,0 +1,640 @@
+/*
+ * Copyright (c) 2022 Pantacor Ltd.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include "logsink.h"
+
+#include <linux/limits.h>
+#include <sys/epoll.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <sys/socket.h>
+#include <sys/wait.h>
+#include <sys/un.h>
+#include <string.h>
+#include <netdb.h>
+#include <stdio.h>
+#include <inttypes.h>
+#include <libgen.h>
+#include <errno.h>
+#include <stdarg.h>
+#include <fcntl.h>
+#include <time.h>
+
+#include "utils/fs.h"
+#include "utils/file.h"
+#include "utils/json.h"
+#include "pvctl_utils.h"
+
+#ifdef DEBUG
+#define WARN_ONCE(msg, args...) 	\
+do {\
+	static bool __warned = false; 	\
+	if (! __warned) { 		\
+		printf(msg, ##args); 	\
+		__warned = true; 	\
+	}\
+}while(0)
+#else
+#define WARN_ONCE(msg, args...)
+
+#endif
+
+
+#include "pantavisor.h"
+#include "buffer.h"
+#include "paths.h"
```

Comment:
I'm afraid you will have to rebase on top of master and check the new functions in paths.h that will allow configurable dynamic paths for the app engine feature. This is applicable for all paths in this PR.